karapace/schema_reader.py

```diff
@@ -17,23 +19,31 @@
 )
 from karapace import constants
 from karapace.config import Config
-from karapace.errors import InvalidSchema
+from karapace.dependency import Dependency
+from karapace.errors import InvalidReferences, InvalidSchema
 from karapace.in_memory_database import InMemoryDatabase
 from karapace.key_format import is_key_in_canonical_format, KeyFormatter, KeyMode
 from karapace.master_coordinator import MasterCoordinator
-from karapace.schema_models import SchemaType, TypedSchema
+from karapace.protobuf.schema import ProtobufSchema
+from karapace.schema_models import parse_protobuf_schema_definition, SchemaType, SchemaVersion, TypedSchema
+from karapace.schema_references import Reference, Referents
 from karapace.statsd import StatsClient
+from karapace.typing import ResolvedVersion
 from karapace.utils import json_decode, JSONDecodeError, KarapaceKafkaClient
 from threading import Condition, Event, Thread
-from typing import Optional
+from typing import Any, Dict, List, Optional, Union
 
+import json
 import logging
 import time
 
 Offset = int
 Subject = str
 Version = int
 
+Schema = Dict[str, Any]
+SchemaId = int
```

Comment:
Redefined from `karapace.typing`. I can't think of a good reason why some of these are redefined in this file!

Reply:
ok